nativelib/src/main/resources/scala-native/time_millis.c

```diff
@@ -11,28 +13,32 @@ long long scalanative_current_time_millis() {
 
 #if defined(_WIN32)
     // January 1, 1970 (start of Unix epoch) in "ticks"
-    long long UNIX_TIME_START = 0x019DB1DED53E8000;
-    long long TICKS_PER_MILLIS = 10000; // a tick is 100ns
+#define UNIX_TIME_START 0x019DB1DED53E8000LL
+#define MILLIS_PER_NANO 1000000
+#define NANOSECONDS_PER_SECOND 1000000000
 
     FILETIME filetime;
     GetSystemTimeAsFileTime(&filetime); // returns ticks in UTC
 
+    int ticksPerMilli =
+        MILLIS_PER_NANO / (NANOSECONDS_PER_SECOND / winFreqQuadPart());
+
     // Copy the low and high parts of FILETIME into a LARGE_INTEGER
     // This is so we can access the full 64-bits as an Int64 without causing
     // an alignment fault
     LARGE_INTEGER li;
     li.LowPart = filetime.dwLowDateTime;
     li.HighPart = filetime.dwHighDateTime;
 
-    current_time_millis = (li.QuadPart - UNIX_TIME_START) / TICKS_PER_MILLIS;
+    current_time_millis = (li.QuadPart - UNIX_TIME_START) / ticksPerMilli;
 #else
-#define MILLIS_PER_SEC 1000LL
-#define MICROS_PER_MILLI 1000LL
+#define MILLIS_PER_SEC 1000
+#define NANOS_PER_MILLI 1000000
 
-    struct timeval tv;
-    gettimeofday(&tv, NULL);
+    struct timespec ts;
+    clock_gettime(CLOCK_REALTIME, &ts);
     current_time_millis =
-        tv.tv_sec * MILLIS_PER_SEC + tv.tv_usec / MICROS_PER_MILLI;
+        ts.tv_sec * MILLIS_PER_SEC + ts.tv_nsec / NANOS_PER_MILLI;
```

Comment:
It is a question of style and author's prerogative. To my aching eyes, introducing parentheses (a * b) + (c / d), makes the 
math stand out easier to verify. 

Similar comment, but less urgent applies in other places.

Reply:
I think more explicit is better than understanding precedent.